### DISCUSSION TITLE
How to pass data across the pages in NextJS without displaying data on URL ?

### DISCUSSION THREAD

**Discussion by ProPhycient5** [vercel/next.js#43846] [Help]
If I need to pass some data from one page to another page, I can do by following way:
```
import Link from 'next/link'

<Link
  href={{
    pathname: '/to-your-other-page',
    query: data // the data
  }}
>
  <a>Some text</a>   
</Link>
```

And access it by

```
import { useRouter } from 'next/router'
const router = useRouter();
const data = router.query;
```

But the issue is that all data are displayed on URL, so how can I pass under the hood those data in `NextJS` without using any `state management library`?

**Comment by ciruz** [comment]
Checkout `Link` with the `as` attribute or `Router`, the `push` method also has an `as` attribute. The `as` attribute is basically the URL what the user will see in his browser.

Be aware this is not available in next 13 - /app directory.

**Reply by ProPhycient5** [reply]
Thanks @ciruz for your reply. But I need the solution for current version of `NextJS` i.e. 13 where I can transfer data across the pages (which is the not child component or page) without showing anything on `URL`.

**Reply by ciruz** [reply]
Ok then you are in the same situation like me. I'm using context / redux right now.

**Reply by ProPhycient5** [reply]
> Ok then you are in the same situation like me. I'm using context / redux right now.

Yeah @ciruz I thought there is any other way to address this issue. Because if we have to transfer less data, it is unnecessary to add state management library in the project.
